plugins/nanospell/plugin.js

```diff
@@ -721,12 +728,19 @@
 			}
 
 			function render(event) {
-				var bookmarks = editor.getSelection().createBookmarks(true),
-					rootElement = event.data;
+				var rootElement = event.data.root,
+					needsBookmarkCreated = event.data.needsBookmarkCreated,
+					bookmarks;
+
+				if (needsBookmarkCreated) {
+					bookmarks = editor.getSelection().createBookmarks(true);
```

Comment:
Might be useful for some future case.  Turns out I don't need it now.  This call is essentially extracted to being wrapped around the loops that fires START_RENDER events.